### ISSUE TITLE
Mismatched package id

### ISSUE THREAD

**Issue by ArchGuyWu** [microsoft/winget-cli#3835]
### Brief description of your issue

I manually installed Windows 10 SDK version 2104, but winget shows the wrong package id(Microsoft.WindowsSDK.10.0.22621), and informs me to upgrade it.

### Steps to reproduce

1. manually install Windows 10 SDK version 2104
2. winget list

### Expected behavior

winget shows the correct package id.

### Actual behavior

winget shows the wrong package id(Microsoft.WindowsSDK.10.0.22621), and informs me to upgrade it.

### Environment

```shell
Windows: Windows.Server v10.0.20348.2031
System Architecture: X64
```

**Comment by denelon** [comment]
@ArchGuyWu which installer was used to install the Windows 10 SDK version 2104?

**Comment by Trenly** [comment]
> @ArchGuyWu which installer was used to install the Windows 10 SDK version 2104?

@denelon - This is mostly related to 
* microsoft/winget-pkgs#124465

and was caused by microsoft/winget-pkgs#123766 not having AppsAndFeaturesEntries. It should be resolved by 
* microsoft/winget-pkgs#124284

**Comment by ArchGuyWu** [comment]
> @ArchGuyWu which installer was used to install the Windows 10 SDK version 2104?

https://go.microsoft.com/fwlink/?linkid=2164145

**Comment by Trenly** [comment]
@ArchGuyWu - - Are you still experiencing this issue with the [latest release](https://github.com/microsoft/winget-cli/releases/latest) of the cli?

**Comment by ArchGuyWu** [comment]
> @ArchGuyWu - - Are you still experiencing this issue with the [latest release](https://github.com/microsoft/winget-cli/releases/latest) of the cli?

Yes, but for Microsoft Visual C++ 2015-2022 Redistributable (x86).

**Comment by denelon** [comment]
We've released WinGet 1.8 with better support for "side-by-side" packages. Is this still happening with the redistributable?
